pallets/loans/src/functions.rs

```diff
@@ -422,62 +512,63 @@ impl<T: Config> Pallet<T> {
 
 	/// accrues rate and debt of a given loan and updates it
 	/// returns the present value of the loan accounting any write offs
-	pub(crate) fn accrue_and_update_loan(
-		pool_id: PoolIdOf<T>,
-		loan_id: T::LoanId,
+	pub(crate) fn accrue_debt_and_calculate_present_value(
+		active_loan: &mut ActiveLoanDetailsOf<T>,
 		write_off_groups: &Vec<WriteOffGroup<T::Rate>>,
 	) -> Result<T::Balance, DispatchError> {
-		Loan::<T>::try_mutate(
-			pool_id,
-			loan_id,
-			|loan| -> Result<T::Balance, DispatchError> {
-				let loan = loan.as_mut().ok_or(Error::<T>::MissingLoan)?;
-
-				// if the loan is not active, then skip updating and return PV as zero
-				if loan.status != LoanStatus::Active {
-					return Ok(Zero::zero());
-				}
-
-				let debt = T::InterestAccrual::current_debt(
-					loan.interest_rate_per_sec,
-					loan.normalized_debt,
-				)?;
+		// TODO: this will do 1 storage read and up to 1 storage write.
+		// Could we optimize this? Or if not, should we return here whether a storage
+		// write was performed (i.e. the debt wasn't updated yet), so that the end
+		// of the update_nav extrinsic, we can only charge for the number of storage
+		// writes that were actually performed.
+		let debt = T::InterestAccrual::current_debt(
+			active_loan.interest_rate_per_sec,
```

Comment:
For other reviewers: this should actually use interest rate + penalty interest rate, but we will implement this later: https://github.com/centrifuge/centrifuge-chain/issues/850